libraries/core/lib/search/metaheuristics/SearchAlgorithm.ts

```diff
@@ -106,75 +99,77 @@ export abstract class SearchAlgorithm<T extends Encoding> {
     budgetManager: BudgetManager<T>,
     terminationManager: TerminationManager
   ): Promise<Archive<T>> {
+    SearchAlgorithm.LOGGER.info("Starting search");
     // Load search subject into the objective manager
     this._objectiveManager.load(subject);
 
     // Start initialization budget tracking
     budgetManager.initializationStarted();
-    // getUserInterface().startProgressBar();
 
     // Inform listeners that the search started
     this._listeners.forEach((listener) => {
       listener.searchStarted(this, budgetManager, terminationManager);
     });
 
-    // getUserInterface().updateProgressBar(
-    //   this.progress("branch"),
-    //   budgetManager.getBudget()
-    // );
-
-    this._eventManager.emitEvent("onSearchInitializationStart");
+    (<TypedEmitter<Events>>process).emit(
+      "searchInitializationStart",
+      this,
+      budgetManager
+    );
 
     // Initialize search process
     await this._initialize(budgetManager, terminationManager);
 
     // Stop initialization budget tracking, inform the listeners, and start search budget tracking
     budgetManager.initializationStopped();
 
-    this._eventManager.emitEvent("onSearchInitializationComplete");
+    (<TypedEmitter<Events>>process).emit(
+      "searchInitializationComplete",
+      this,
+      budgetManager
+    );
 
     this._listeners.forEach((listener) =>
       listener.initializationDone(this, budgetManager, terminationManager)
     );
 
-    // getUserInterface().updateProgressBar(
-    //   this.progress("branch"),
-    //   budgetManager.getBudget()
-    // );
     budgetManager.searchStarted();
 
-    this._eventManager.emitEvent("onSearchStart");
+    (<TypedEmitter<Events>>process).emit("searchStart", this, budgetManager);
 
     // Start search until the budget has expired, a termination trigger has been triggered, or there are no more objectives
     while (
       this._objectiveManager.hasObjectives() &&
       budgetManager.hasBudgetLeft() &&
       !terminationManager.isTriggered()
     ) {
-      this._eventManager.emitEvent("onSearchIterationStart");
+      (<TypedEmitter<Events>>process).emit(
+        "searchIterationStart",
+        this,
+        budgetManager
+      );
 
       // Start next iteration of the search process
       await this._iterate(budgetManager, terminationManager);
 
       // Inform the budget manager and listeners that an iteration happened
       budgetManager.iteration(this);
 
-      this._eventManager.emitEvent("onSearchIterationComplete");
+      (<TypedEmitter<Events>>process).emit(
```

Comment:
We could maybe remove one of these events as we don't need an event as the start and end of a while loop

Reply:
After a discussion we decided that it cannot hurt to keep both events in here.